File: dashboard/src/types/workspace.ts
export interface Workspace {
  id: string;
  organization_id: string;
  name: string;
  url: string;
  slug: string;
  api_key: string;
  allowed_origins: string[];
  tags: string[];
}

export interface WorkspaceCreate {
  name: string;
  url: string;
  slug: string;
}

export interface WorkspaceUpdate {
  name?: string;
  url?: string;
  slug?: string;
  allowed_origins?: string[];
  regenerate_api_key?: boolean;
}

File: api/app/schemas/workspace.py
from pydantic import BaseModel, ConfigDict, Field, field_validator
from uuid import UUID
from typing import Optional, List

#############################################################################
#############################################################################
class WorkspaceCreate(BaseModel):
    name: str
    url: str
    slug: str
    allowed_origins: List[str] = []

    @field_validator("slug")
    @classmethod
    def slug_to_lower(cls, v: str) -> str:
        return v.lower().strip()

#############################################################################
#############################################################################
class WorkspaceUpdate(BaseModel):
    name: Optional[str] = None
    url: Optional[str] = None
    slug: Optional[str] = None
    allowed_origins: Optional[List[str]] = None

    regenerate_api_key: Optional[bool] = Field(default=False)

    @field_validator("slug")
    @classmethod
    def slug_to_lower(cls, v: Optional[str]) -> Optional[str]:
        if v is not None:
            return v.lower().strip()
        return v

#############################################################################
#############################################################################
class WorkspaceResponse(BaseModel):
    id: UUID
    organization_id: UUID
    name: str
    url: str
    slug: str

    api_key: str
    allowed_origins: List[str] = []

    model_config = ConfigDict(from_attributes=True)